### PR TITLE
[#165028951] update italia ts common with errors support

### PR DESCRIPTION
**Pull request by Undermaken**
wip refactoring

### CODE REVIEW

File: ts/boot/configureStoreAndPersistor.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ import {
 } from "../store/reducers";
 import { getInitialState as getInstallationInitialState } from "../store/reducers/notifications/installation";
 import { GlobalState, PersistedGlobalState } from "../store/reducers/types";
+import { PotTransform } from "../store/transforms/potTransforms";
```
Comment (cloudify) [inline ts/boot/configureStoreAndPersistor.ts]: this change shouldn't be in this PR - @fpersico ?

File: ts/sagas/backendInfo.ts [hunk 2]
```diff
@@ -22,11 +22,16 @@ const BACKEND_INFO_RETRY_INTERVAL = 60 * 60 * 10 * 1000;
 function* backendInfoWatcher(): IterableIterator<Effect> {
   const backendPublicClient = BackendPublicClient(apiUrlPrefix);
 
-  function getServerInfo(): Promise<BasicResponseType<ServerInfo> | undefined> {
+  function getServerInfo(): Promise<
+    Either<Error, BasicResponseType<ServerInfo>>
```
Comment (cloudify) [inline ts/sagas/backendInfo.ts]: this should probably be `t.Validation<BasicResponseType<ServerInfo> | undefined>` where `t` is `import * as t from "io-ts";`

see [here](https://github.com/teamdigitale/io-ts-commons/blob/master/src/requests.ts#L105)

File: ts/sagas/backendInfo.ts [hunk 3]
```diff
@@ -22,11 +22,16 @@ const BACKEND_INFO_RETRY_INTERVAL = 60 * 60 * 10 * 1000;
...
+  > {
     return new Promise((resolve, _) =>
-      backendPublicClient
-        .getServerInfo({})
-        .then(resolve, () => resolve(undefined))
+      backendPublicClient.getServerInfo({}).then(
+        x => {
+          return x;
```
Comment (cloudify) [inline ts/sagas/backendInfo.ts]: you should call `resolve` here, the return value of this lambda gets discarded

File: ts/boot/configureStoreAndPersistor.ts (hunk 1 above)
Comment (fpersico) [reply ts/boot/configureStoreAndPersistor.ts]: Don't know why it is here. I think @Undermaken is testing if this PR fixes the errors in the other story?

File: ts/sagas/contentLoaders.ts [hunk 4]
```diff
@@ -19,11 +19,14 @@ const contentClient = ContentClient();
  */
 function getServiceMetadata(
   serviceId: ServiceId
-): Promise<BasicResponseType<ServiceMetadata> | undefined> {
+): Promise<Either<Error, BasicResponseType<ServiceMetadata>>> {
```
Comment (cloudify) [inline ts/sagas/contentLoaders.ts]: also make this `t.Validation< ServiceMetadata > | undefined` (see https://github.com/teamdigitale/io-ts-commons/blob/master/src/requests.ts#L105)

File: ts/boot/configureStoreAndPersistor.ts (hunk 1 above)
Comment (Undermaken) [reply ts/boot/configureStoreAndPersistor.ts]: I don't know too. I created this branch starting from master and I can't figure how these changes are included.

**Comment by digitalcitizenship** [comment]
## Affected stories

  * 🌟 [#165028951](https://www.pivotaltracker.com/story/show/165028951): Aggiornare italia-ts-commons con il supporto agli errori di validazione delle richieste

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request closed without merging**